core/storage/user/gae_models.py

```diff
@@ -2214,3 +2214,66 @@ def get_new_id(cls, unused_entity_name):
                 return new_id
 
         raise Exception('New id generator is producing too many collisions.')
+
+
+class UserAuthModel(base_models.BaseModel):
+    """Stores the authentication details for a particular user.
+
+    Instances of this class are keyed by user id.
+    """
+
+    # Authentication detail for sign in using google id (GAE).
+    gae_id = ndb.StringProperty(indexed=True, repeated=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """The model can be deleted since it only contains information
+        relevant to one user account.
+        """
+        return base_models.DELETION_POLICY.DELETE
+
+    @staticmethod
+    def get_export_policy():
+        """Model does not need to exported as it hold authentication details
```

Comment:
Nit:

```
Model does not need to be exported as it holds authentication details
relevant only for the backend, and does not contain any exportable
user data.
```

Reply:
Also, is this always true? Won't we someday store PII here for authentication purposes? I actually think we do need to export the data here, even if the current ID is not specifically useful to the user.

@varun-tandon what are your thoughts? Should we not export this model now, and update the export policy later once we actually have data to export?

Reply:
Hmm. This is a little tricky. I think that the ideal situation is that we set the export_policy as not applicable here since there is currently no PII, but once the PII is added, we update the export policy. I would be a little wary of that situation however, since it is possible that a reviewer doesn't notice that we need to update the export policy and export data. @BenHenning do you think there is some way we can ensure that reviewers are looking out for addition of PII in a Takeout context?

Reply:
@varun-tandon I might not have understood fully what you said, but do you want me to add a comment in the method saying "In future change the export policy to CONTAINS_USER_DATA if we are storing PII in the model".

Reply:
I'm not sure if that would solve the issue. I think @seanlip would be the best person to ask about this.

Reply:
Actually it's a question for @BenHenning.

Reply:
Hmm. One way to do this is to validate all existing fields (eg via a test) and if a new field is added that test would fail. This might be something we’d want to do for all models marked as not applicable. 

That being said, we do have storage model codeowners for this very reason. I think we can trust reviewers here, but automation is better since it’s less likely to make a mistake. Is the approach above feasible?

Reply:
OK, sounds good. We can send this PR in, and after my one-off-job is complete, I can implement some testing on a property level.

Reply:
That sounds great--thanks @varun-tandon!

@sajalasati I think it's worth mentioning in the comment that this model may contain future user data in the interim until the new tests are in place.

Reply:
@BenHenning Changed the function documentation now.